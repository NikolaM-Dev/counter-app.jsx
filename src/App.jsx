export const HelloWorldApp = () => {
  return <h1>Hello World</h1>;
};
