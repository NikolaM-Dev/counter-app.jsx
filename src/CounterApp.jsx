import PropTypes from 'prop-types';
import { useState } from 'react';

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    setCounter(counter + 1);
  };

  const handleReset = () => {
    setCounter(value);
  };

  const handleSubstraction = () => {
    setCounter(counter - 1);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>

      <button onClick={handleSubstraction}>-1</button>
      <button onClick={handleReset}>reset</button>
      <button onClick={handleAdd}>+1</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
