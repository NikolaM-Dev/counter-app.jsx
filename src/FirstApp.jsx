import PropTypes from 'prop-types';

export const FirstApp = ({ title, subTitle }) => {
  return (
    <>
      <h1>{title}</h1>
      <h2>{subTitle}</h2>
      <p>I&apos;m a subtitle</p>
    </>
  );
};

FirstApp.propTypes = {
  subTitle: PropTypes.number,
  title: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  subTitle: 'Dragon Ball Z',
  title: "Hello, I'm Nikola",
};
