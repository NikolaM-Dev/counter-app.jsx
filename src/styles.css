html,
body {
  background-color: #21232a;
  color: white;
  font-family: system-ui;
  font-size: 1.3rem;
  padding: 70px;
}
