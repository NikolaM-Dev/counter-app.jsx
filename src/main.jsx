import React from 'react';
import ReactDOM from 'react-dom/client';

import { HelloWorldApp } from './App';
  return <h1>Hello World</h1>;
};

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <HelloWorldApp />
  </React.StrictMode>,
);
