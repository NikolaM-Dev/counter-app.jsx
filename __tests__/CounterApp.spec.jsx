import { fireEvent, render, screen } from '@testing-library/react';

import { CounterApp } from '../src/CounterApp';

describe('Tests on <CounterApp />', () => {
  const initialValue = 10;

  it('should match the snapshot', () => {
    const { container } = render(<CounterApp value={initialValue} />);

    expect(container).toMatchSnapshot();
  });

  it('should to show an initial value of 100', () => {
    const initialValue = 100;

    render(<CounterApp value={initialValue} />);

    expect(screen.getByText(initialValue)).toBeTruthy();
  });

  it('should increment with the +1 button', () => {
    render(<CounterApp value={initialValue} />);

    const btnAdd = screen.getByText('+1');

    fireEvent.click(btnAdd);

    expect(screen.getByText(11)).toBeTruthy();
  });

  it('should decrement with the -1 button', () => {
    render(<CounterApp value={initialValue} />);

    const btnSubstrac = screen.getByText('-1');

    fireEvent.click(btnSubstrac);

    expect(screen.getByText(9)).toBeTruthy();
  });

  it('should return to initial value with reset button', () => {
    render(<CounterApp value={initialValue} />);

    const btnAdd = screen.getByText('+1');
    const btnReset = screen.getByRole('button', { name: 'btn-reset' });

    fireEvent.click(btnAdd);
    fireEvent.click(btnAdd);
    fireEvent.click(btnReset);

    expect(screen.getByText(initialValue)).toBeTruthy();
  });
});
