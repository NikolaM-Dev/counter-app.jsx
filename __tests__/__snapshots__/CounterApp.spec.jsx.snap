// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <CounterApp /> should match the snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    reset
  </button>
  <button>
    +1
  </button>
</div>
`;
